packages/babel-traverse/test/scope.js

```diff
@@ -902,5 +902,30 @@ describe("scope", () => {
       `);
       expect(program.scope.hasOwnBinding("class")).toBe(true);
     });
+    it("registers the new binding outside function when the path is a param initializer", () => {
+      const program = getPath("(a = f()) => {}");
+      const assignmentPattern = program.get("body.0.expression.params.0");
+      assignmentPattern.scope.push({ id: t.identifier("ref") });
+      expect(program.toString()).toMatchInlineSnapshot(`
+        "var ref;
```

Comment:
`var ref` is created by `assignmentPattern.scope.push({ id: t.identifier("ref") });`, this test is to show that the `ref` is pushed to the parent scope instead of
```js
class C {
  m(a = f()) { var ref; }
}
```